src/application.rs

```diff
@@ -68,17 +68,20 @@ impl Application {
         ui::draw(&mut self.terminal, &state);
 
         let (_, server_addr) = self.network.listen_tcp("0.0.0.0:0").unwrap();
+        let server_port = server_addr.port();
+        
         self.network.listen_udp_multicast(self.discovery_addr).unwrap();
 
         let discovery_endpoint = self.network.connect_udp(self.discovery_addr).unwrap();
-        self.network.send(discovery_endpoint, NetMessage::HelloLan(self.user_name.clone(), server_addr)).unwrap();
+        self.network.send(discovery_endpoint, NetMessage::HelloLan(self.user_name.clone(), server_port)).unwrap();
 
         loop {
             match self.event_queue.receive() {
                 Event::Network(net_event) => match net_event {
                     NetEvent::Message(endpoint, message) => match message {
                         // by udp (multicast):
-                        NetMessage::HelloLan(user, server_addr) => {
+                        NetMessage::HelloLan(user, server_port) => {
+							let server_addr = format!("{}:{}", endpoint.addr().ip(), server_port);
```

Comment:
There is no need to transform it into a string to transform it later into a SocketAddr. It can be directly a socket address. If I remember well, a tuple (ip, port) is transformable into a SocketAddr. 

The rest of the changes are perfect!

Reply:
You're right. done